homeassistant/core.py

```diff
@@ -1184,7 +1193,12 @@ def async_set(
         This method must be run in the event loop.
         """
         entity_id = entity_id.lower()
-        new_state = str(new_state)
+        if isinstance(new_state, float):
```

Comment:
We convert the state to a string already in the Entity helper. The other significant place where this method is used is in the http api.

Reply:
Thanks Martin, you're absolutely right.
I'll rework this + add a test case on entity level.